Warehouse FAQ updates (#660)
rudderlabs/rudderstack-docs: docs/data-warehouse-integrations/warehouse-faqs.mdx

You can view the warehouse upload status in the **Live Events** section of the destination in the RudderStack dashboard.

## Staging

> ### Why am I observing files written in the location `<namespace>/rudder-warehouse-staging-logs/`?

Reviewer: ```suggestion
### What are these files being written in the location `<namespace>/rudder-warehouse-staging-logs/`?
```

You can view the warehouse upload status in the **Live Events** section of the destination in the RudderStack dashboard.

## Staging

### Why am I observing files written in the location `<namespace>/rudder-warehouse-staging-logs/`?

> RudderStack collects all the unprocessed data flowing to the warehouse destinations as staging files. It stores these files in the object storage at the location `/rudder-warehouse-staging-logs/`. 

Reviewer 2: Do you think we should add location as `<namespace>/rudder-warehouse-staging-logs/`

> RudderStack collects all the unprocessed data flowing to the warehouse destinations as staging files. It stores these files in the object storage at the location `/rudder-warehouse-staging-logs/`. 

Author: ```suggestion
RudderStack collects all the unprocessed data flowing to the warehouse destinations as staging files. It stores these files in the object storage at the location `<namespace>/rudder-warehouse-staging-logs/`. 
```

- For **Data Lake** destinations (**S3 Data Lake, GCS Data Lake, Azure Data Lake**), there is no set limit.
- For **other** destinations, the table name is truncated after 127 characters.

### Can I change the name of the table in which my events are synced?

> Yes, you can add a <Link to="">transformation</Link> to change the event name and, consequently, the name of the table in which the events are synced. 

Reviewer: hyperlink to be added

> Yes, you can add a <Link to="">transformation</Link> to change the event name and, consequently, the name of the table in which the events are synced. 

Author: ```suggestion
Yes, you can add a <Link to="/transformations/">transformation</Link> to change the event name and, consequently, the name of the table in which the events are synced. 
```

products_android_version : 16
```

### How does RudderStack determine the column data type? Can I change an existing data type for a column?

> RudderStack determines the columns' data type based on the values contained in the first event when it is uploaded (during the first upload sync).

Reviewer: ```suggestion
RudderStack determines the data type of a column based on its value in the first event (during the first upload sync).
```

products_android_version : 16
```

### How does RudderStack determine the column data type? Can I change an existing data type for a column?

RudderStack determines the columns' data type based on the values contained in the event when it is uploaded.

For example, suppose `column_x` is received with the value as `1`. RudderStack then sets the data type of this column as `int` in the event table.

<div class="infoBlock">
Although you can change the columns' data type in the warehouse any time, the changes will be applicable to the events from the next sync.
</div>

To set your preferred data type for a particular column, it is highly recommended to follow these steps:

1. Create a column in the warehouse with a dummy name and the required data type.
2. Cast the data from the original column and load it into the dummy column.
3. Drop the original column.
4. Rename the dummy column to the original column name.

<div class="dangerBlock">

> During steps 3 and 4, the tables will be in a locked state. This might impact real-time data uploads/syncs.

Reviewer 2: Please mention this option: 
`If above operations are taking longer time, you can halt warehouse operations for time being. More at: https://www.rudderstack.com/docs/data-warehouse-integrations/warehouse-faqs/#how-do-i-avoid-lags-in-data-syncs-while-heavy-loads-are-running-on-my-datawarehouse`
